onnxruntime/python/onnxruntime_pybind_ortvalue.cc

```diff
@@ -421,21 +421,39 @@ void addOrtValueMethods(pybind11::module& m) {
       // Converts Tensor into a numpy array
       .def("numpy", [](const OrtValue* ml_value) -> py::object {
         ORT_ENFORCE(ml_value->IsTensor(), "Only OrtValues that are Tensors are convertible to Numpy objects");
-
+        const auto& device = ml_value->Get<Tensor>().Location().device;
```

Comment:
Based on the build failures in #25683 this might be needed. Do you want to merge that changes from that PR into here so everything goes in at once?

```suggestion
        [[maybe_unused]] const auto& device = ml_value->Get<Tensor>().Location().device;
```